tools/util_windows.go

```diff
@@ -0,0 +1,163 @@
+// +build windows
+
+package tools
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"unsafe"
+
+	"github.com/rubyist/tracerx"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	availableClusterSize = []int64{64 * 1024, 4 * 1024} // ReFS only supports 64KiB and 4KiB cluster.
+	GiB                  = int64(1024 * 1024 * 1024)
+)
+
+// _FSCTL_DUPLICATE_EXTENTS_TO_FILE :
+// Instructs the file system to copy a range of file bytes on behalf of an application.
+//
+// https://docs.microsoft.com/windows/win32/api/winioctl/ni-winioctl-fsctl_duplicate_extents_to_file
+const _FSCTL_DUPLICATE_EXTENTS_TO_FILE = 623428
+
+// _DUPLICATE_EXTENTS_DATA :
+// Contains parameters for the FSCTL_DUPLICATE_EXTENTS control code that performs the Block Cloning operation.
+//
+// https://docs.microsoft.com/windows/win32/api/winioctl/ns-winioctl-duplicate_extents_data
+type _DUPLICATE_EXTENTS_DATA struct {
```

Comment:
Same issue about the leading underscore.  Since this is not supposed to be accessible outside of this package, we probably want to call it `duplicateExtentsData`.

Is there a way for us to get the definition of this struct and constant from a library so that we don't have to implement it ourselves?